### PR TITLE
Removed redundant c_str() calls

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary

```SUMMARY: Infrastructure "Removed redundant c_str calls"```

#### Purpose of change

Cleanup code from excessive calls to `c_str()` function to increase readbility.

#### Additional context

I've removed 2275 our of 3102 occurrences of `.c_str()` from 183 files (827 occurrences are still remaining in 125 files), so changes are pretty big and I can split changes across several commits/pull requests if necessary, but want to make sure it compiles fine on CI beforehand.

### CODE REVIEW

**Comment by mlangsdorf** [comment]
You're the one who gets to resolve the merge conflicts, so thanks for doing this.

Reviewed basecamp, faction_camp, dialogue_window, mission*, npc*, and veh* and everything looks good.

**Comment by ZhilkinSerg** [comment]
Jenkins fails due to changes to `json.h`, otherwise should be good to go for reviewing.

I can split to several chunks for easier reviewing and merging if necessary.

File: src/player.cpp [hunk 1]
```diff
@@ -3002,7 +3002,7 @@ void player::search_surroundings()
               tr.loadid == tr_landmine_buried || tr.loadid == tr_sinkhole ) ) {
             const std::string direction = direction_name( direction_from( pos(), tp ) );
             add_msg_if_player( m_warning, _( "Your ground sonar detected a %1$s to the %2$s!" ),
-                               tr.name().c_str(), direction.c_str() );
+                               tr.name(), direction.c_str() );
```
Comment (ifreund) [inline src/player.cpp]:
```suggestion
                               tr.name(), direction );
```

File: src/player.cpp [hunk 2]
```diff
@@ -3019,7 +3019,7 @@ void player::search_surroundings()
                 const std::string direction = direction_name(
                                                   direction_from( pos(), tp ) );
                 add_msg_if_player( _( "You've spotted a %1$s to the %2$s!" ),
-                                   tr.name().c_str(), direction.c_str() );
+                                   tr.name(), direction.c_str() );
```
Comment (ifreund) [inline src/player.cpp]:
```suggestion
                                   tr.name(), direction );
```

**Comment by kevingranade** [comment]
Considering the nature of the change, I just want to fire the whole thing off.

**Pull request merged**